core/src/price_estimation/clients/apiclient.rs

```diff
@@ -1,79 +1,88 @@
-mod api;
-
-use super::{PriceSource, TokenData};
+use super::super::{PriceSource, TokenData};
 use crate::http::HttpFactory;
 use crate::models::TokenId;
 use anyhow::{anyhow, Context, Result};
-use api::{DexagApi, DexagHttpApi};
 use futures::{
     future::{self, BoxFuture, FutureExt as _},
     lock::Mutex,
 };
 use std::collections::HashMap;
 
-struct ApiTokens {
+pub trait Symbolic {
```

Comment:
Since Token is only data I wouldn't mock it but instead use the plain old data model and create dummy objects in the tests. We only really need mocks for components that contain logic.

Reply:
I'm not sure I understand the suggestion, maybe I don't understand the word "data model". Is it different from my manually implemented `MockSymbolic`?

Reply:
So if a struct has an Impl with lots of logic, I think it's a very good idea to mock it. However if a struct is just about the data it carries and the unit under test doesn't call any functions on it other than accessing its members, we don't need to mock and can just use an instantiation of this "plain old data object" itself (https://en.wikipedia.org/wiki/Passive_data_structure)